merino/providers/adm.py

```diff
@@ -182,14 +186,25 @@ async def _fetch(self) -> None:
             self.backend.fetch_attachment(item["attachment"]["location"])
             for item in records
         ]
+        fkw_index = 0
         for done_task in as_completed(fetch_tasks):
             res = await done_task
+
             for suggestion in res.json():
-                id = len(results)
-                for kw in suggestion.pop("keywords", []):
-                    # Note that for adM suggestions, each keyword can only be mapped to
-                    # a single suggestion.
-                    suggestions[kw] = id
+                result_id = len(results)
+                keywords = suggestion.pop("keywords", [])
+                full_keywords_tuples = suggestion.pop("full_keywords", [])
+                begin = 0
+                for full_keyword, n in full_keywords_tuples:
+                    for query, fkw in zip(
+                        islice(keywords, begin, begin + n), repeat(full_keyword, n)
+                    ):
```

Comment:
`fkw` isn't used so I don't think the `zip` necessary? also `keywords` should always be a list so do we need `islice`?
```suggestion
                    for query in keywords[begin : begin + n]:
```